drivers/ieee802154/ieee802154_efr32.c

```diff
@@ -0,0 +1,558 @@
+
+    net_if_set_link_addr(iface, mac, 8, NET_LINK_IEEE802154);
+    efr32->iface = iface;
+    ieee802154_init(iface);
+}
+
+static enum ieee802154_hw_caps efr32_get_capabilities(struct device *dev)
+{
+    return IEEE802154_HW_FCS |
+           IEEE802154_HW_2_4_GHZ |
+           IEEE802154_HW_FILTER |
+           IEEE802154_HW_TX_RX_ACK;
+}
+
+static int efr32_cca(struct device *dev)
+{
+    LOG_DBG("Doing CCA");
+}
+
+static int efr32_set_channel(struct device *dev, u16_t channel)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_PrepareChannel(efr32->rail_handle, channel);
+
+    if (status != RAIL_STATUS_NO_ERROR)
+    {
+        return -EIO;
+    }
+
+    efr32->channel = channel;
+
+    return 0;
+}
+
+static int efr32_set_pan_id(struct device *dev, u16_t pan_id)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_IEEE802154_SetPanId(efr32->rail_handle, pan_id, 0);
+
+    if (status != RAIL_STATUS_NO_ERROR)
+    {
+        return -EIO;
+    }
+
+    return 0;
+}
+
+static int efr32_set_short_addr(struct device *dev, u16_t short_addr)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_IEEE802154_SetShortAddress(efr32->rail_handle, short_addr, 0);
+
+    if (status != RAIL_STATUS_NO_ERROR)
+    {
+        return -EIO;
+    }
+
+    return 0;
+}
+
+static int efr32_set_ieee_addr(struct device *dev, const u8_t *ieee_addr)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_IEEE802154_SetLongAddress(efr32->rail_handle, ieee_addr, 0);
+
+    if (status != RAIL_STATUS_NO_ERROR)
+    {
+        LOG_ERR("Error setting address via RAIL");
+        return -EIO;
+    }
+
+    return 0;
+}
+
+static int efr32_filter(struct device *dev,
+                        bool set,
+                        enum ieee802154_filter_type type,
+                        const struct ieee802154_filter *filter)
+{
+    LOG_DBG("Applying filter %u", type);
+
+    if (!set)
+    {
+        return -ENOTSUP;
+    }
+
+    if (type == IEEE802154_FILTER_TYPE_IEEE_ADDR)
+    {
+        return efr32_set_ieee_addr(dev, filter->ieee_addr);
+    }
+    else if (type == IEEE802154_FILTER_TYPE_SHORT_ADDR)
+    {
+        return efr32_set_short_addr(dev, filter->short_addr);
+    }
+    else if (type == IEEE802154_FILTER_TYPE_PAN_ID)
+    {
+        return efr32_set_pan_id(dev, filter->pan_id);
+    }
+
+    return -ENOTSUP;
+}
+
+static int efr32_set_txpower(struct device *dev, s16_t dbm)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_SetTxPower(efr32->rail_handle, dbm);
+
+    if (status != RAIL_STATUS_NO_ERROR)
+    {
+        return -EIO;
+    }
+
+    return 0;
+}
+
+static int efr32_start(struct device *dev)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+    ARG_UNUSED(efr32);
```

Comment:
if you don't need it, don't take it from dev. Just ARG_UNUSED(dev)   apply that everywhere where relevant